From 3fd4db2ce1710f45eb297ae6b2c10726b33204f3 Mon Sep 17 00:00:00 2001
From: Andrea Adami <andrea.adami@gmail.com>
Date: Sun, 29 Apr 2018 00:46:16 +0200
Subject: [PATCH] arm64: crashdump-arm64.c: fix warning

                 from ../git/kexec/kexec.h:6,
                 from ../git/kexec/arch/arm64/crashdump-arm64.c:18:
/tmp/build/tmp-musl/work/aarch64-oe-linux-musl/kexec-tools-klibc/0.6+gitAUTOINC+
0481e9ed61-r0/recipe-sysroot/usr/lib/klibc/include/klibc/compiler.h:144:0: warni
ng: "__bitwise" redefined
 # define __bitwise

Signed-off-by: Andrea Adami <andrea.adami@gmail.com>

---
 kexec/arch/arm64/crashdump-arm64.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kexec/arch/arm64/crashdump-arm64.c b/kexec/arch/arm64/crashdump-arm64.c
index 4fd7aa8..890d88f 100644
--- a/kexec/arch/arm64/crashdump-arm64.c
+++ b/kexec/arch/arm64/crashdump-arm64.c
@@ -13,6 +13,9 @@
 #define _GNU_SOURCE
 
 #include <errno.h>
+#ifdef __KLIBC__
+#include <sys/types.h>
+#endif
 #include <linux/elf.h>
 
 #include "kexec.h"
